File: trunk/server/src/main/java/org/ngame/protocol/battle/BaseActionResponse.java
package org.ngame.protocol.battle;

import lombok.Data;

import java.util.List;

import org.ngame.protocol.BaseResponse;
import org.ngame.protocol.annotation.Proto;
import org.ngame.protocol.domain.SkillInfo;

import com.baidu.bjf.remoting.protobuf.annotation.Protobuf;

/**
 *
 * @author beykery
 */
@Data
@Proto(value = -24, description = "基地行动")
public class BaseActionResponse extends BaseResponse
{
   @Protobuf(description = "用户id")
   private String pid;
   @Protobuf(description = "技能")
   private List<SkillInfo> skill;
}

File: trunk/unityClientSample/Assets/src/protocol/BaseActionResponse.cs
using ProtoBuf;
using System.Collections.Generic;
using System;

namespace protocol
{
    /// <summary>
    /// 基地行动 协议:-24
    /// </summary>
	[Proto(value=-24,description="基地行动")]
	[ProtoContract]
	public class BaseActionResponse
	{
        /// <summary>
        ///  用户id
        /// </summary>
		[ProtoMember(3, IsRequired = false)]
		public string pid;
        /// <summary>
        ///  技能
        /// </summary>
		[ProtoMember(4, IsRequired = false)]
		public List<SkillInfo> skill;
        /// <summary>
        ///  是否成功
        /// </summary>
		[ProtoMember(1, IsRequired = false)]
		public bool success;
        /// <summary>
        ///  错误消息
        /// </summary>
		[ProtoMember(2, IsRequired = false)]
		public string info;

	}
}